scripts/v.db.join/v.db.join.py

```diff
@@ -173,28 +183,41 @@ def main():
 
         # add only the new column to the table
         if colname.lower() not in all_cols_tt:
-            try:
-                grass.run_command(
-                    "v.db.addcolumn", map=map, columns=colspec, layer=layer
-                )
-            except CalledModuleError:
-                grass.fatal(_("Error creating column <%s>") % colname)
-
-        stmt = template.substitute(
-            table=maptable,
-            column=column,
-            otable=otable,
-            ocolumn=ocolumn,
-            colname=colname,
+            cols_to_add_final.append(colspec)
+
+    cols_added = [col.split(" ")[0] for col in cols_to_add_final]
+    cols_added_str = ",".join(cols_added)
+    try:
+        grass.run_command(
+            "v.db.addcolumn", map=map, columns=cols_to_add_final, layer=layer
+        )
+    except CalledModuleError:
+        grass.fatal(_("Error creating columns <%s>") % cols_added_str)
+
+    update_str = "BEGIN TRANSACTION\n"
+    for col in cols_added:
+        cur_up_str = (
+            f"UPDATE {maptable} SET {col} = (SELECT {col} FROM "
+            f"{otable} WHERE "
+            f"{otable}.{ocolumn}={maptable}.{column});\n"
         )
-        grass.debug(stmt, 1)
-        grass.verbose(_("Updating column <%s> of vector map <%s>...") % (colname, map))
-        try:
-            grass.write_command(
-                "db.execute", stdin=stmt, input="-", database=database, driver=driver
-            )
-        except CalledModuleError:
-            grass.fatal(_("Error filling column <%s>") % colname)
+        update_str += cur_up_str
+    update_str += "END TRANSACTION"
+    grass.debug(update_str, 1)
+    grass.verbose(_(f"Updating columns {cols_added_str} of vector map {map}..."))
+    sql_file = grass.tempfile()
+    rm_files.append(sql_file)
+    with open(sql_file, "w") as write_file:
+        write_file.write(update_str)
+    try:
+        grass.run_command(
+            "db.execute",
+            input=sql_file,
+            database=database,
+            driver=driver,
+        )
+    except CalledModuleError:
+        grass.fatal(_(f"Error filling columns {cols_added_str}"))
```

Comment:
For trans string use `.format()` method instead of f-string, please.

```python
grass.fatal(_("Error filling columns {}").format(cols_added_str))
```